airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/AirbyteValueMapper.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.data
+
+interface AirbyteValueMapper {
+    fun collectFailure(path: List<String>)
+    fun map(
+        value: AirbyteValue,
+        schema: AirbyteType,
+        path: List<String> = emptyList()
+    ): AirbyteValue =
+        try {
+            when (schema) {
+                is ObjectType -> mapObject(value as ObjectValue, schema, path)
+                is ObjectTypeWithoutSchema ->
+                    mapObjectWithoutSchema(value as ObjectValue, schema, path)
+                is ObjectTypeWithEmptySchema ->
+                    mapObjectWithEmptySchema(value as ObjectValue, schema, path)
+                is ArrayType -> mapArray(value as ArrayValue, schema, path)
+                is ArrayTypeWithoutSchema ->
+                    mapArrayWithoutSchema(value as ArrayValue, schema, path)
+                is UnionType -> mapUnion(value, schema, path)
+                is BooleanType -> mapBoolean(value as BooleanValue, path)
+                is NumberType -> mapNumber(value as NumberValue, path)
+                is StringType -> mapString(value as StringValue, path)
+                is IntegerType -> mapInteger(value as IntegerValue, path)
+                is DateType -> mapDate(value as DateValue, path)
+                is TimeTypeWithTimezone -> mapTimeWithTimezone(value as TimeValue, path)
+                is TimeTypeWithoutTimezone -> mapTimeWithoutTimezone(value as TimeValue, path)
+                is TimestampTypeWithTimezone ->
+                    mapTimestampWithTimezone(value as TimestampValue, path)
+                is TimestampTypeWithoutTimezone ->
+                    mapTimestampWithoutTimezone(value as TimestampValue, path)
+                is NullType -> mapNull(path)
+                is UnknownType -> mapUnknown(value as UnknownValue, path)
+            }
+        } catch (e: Exception) {
```

Comment:
doublechecking: this is most likely some sort of "cannot cast StringValue to IntValue" exception?

Reply:
I'm assuming this is a catchall for any uncaught exception. Normal usage should be to explicitly look for cases that need nulling and use `collectFailure` explicitly?

Reply:
ah yeah that makes sense. I was just checking that with the code that currently exists - the only way we're running into this catch handler is if we fail to cast? (i.e. none of the methods are doing any interesting `Instant.parse` / etc)

Reply:
since right now any marshaling exceptions end up as unknown type, it probably makes sense to turn unknown type into a null here. I'll go ahead and do that.

Reply:
maybe better to just pass through? I.e. if we don't recognize a field type
* default to "widest type" in the destination
* pass through whatever value we get

which I think will make our lives easier if we want to add new data types (variant :P ), since then the rollout is easy:
* sources update their catalog to use the new type
* destinations add support for the new type

as opposed to if we null out unrecognized types, then there's some complicated dance to avoid losing data in weird ways

Reply:
gotcha, I'll pass through for now